### DISCUSSION TITLE
How to make abp generate-proxy to generate classes instead of interfaces?

### DISCUSSION THREAD

**Discussion by gunpal5** [abpframework/abp#9435] [Q&A] [answered]
Hello,

I was trying to generate some Service Proxies with abp generate-proxy command, but it only generates interfaces for the Dto Models like this:
![image](https://user-images.githubusercontent.com/10114874/123460720-af59a380-d605-11eb-954b-5ce360d0e11f.png)

I need these models to be generated as classes not interface, so that I can initialize class objects in codes. Is there any way to do this?

Regards,
Gunpal Jain

**Comment by amsaighi** [comment, marked as answer]
The abp generate-proxy generate only Models, Services and Enum. check the official docs  [Service Proxies](https://docs.abp.io/en/abp/latest/UI/Angular/Service-Proxies)

**Reply by gunpal5** [reply]
Thanks, I figured out how to initialize the DTO interface in typescript for the service calls. 

like this:
inputDto:SomeDto = {} as SomeDto;

instead of inputDto = new SomeDto();
